### ISSUE TITLE
Deploy to Kubernetes: Service conversion isn't very good with multi-container environments.

### ISSUE THREAD

**Issue by cdrage** [containers/podman-desktop#4725]
### Is your enhancement related to a problem? Please describe

I ran into another issue with converting to Kubernetes, to be honest the conversion from the group of compose containers to Kubernetes is done.. poorly with regards to services and being able to access them through the namespace.

For example, if we were to deploy only ONE container, it would be converted correctly. But say if we were to deploy multiple containers where they would be able to access each-other via services? For example, within a `frontend` container, you can ping `redis-master`, it will not work due to the way we are converting services.

When converting this example:

```yaml
services:

  redis-master:
    container_name: redis-master
    image: redis:latest
    ports:
      - "6379"

  redis-replica:
    container_name: redis-replica
    image: redis:latest
    ports:
      - "6379"
    command: redis-server --replicaof redis-master 6379
    depends_on:
      - redis-master

  frontend:
    container_name: frontend
    image: cdrage/frontend
    #build: ./frontend
    ports:
      - "8080:8080"
```

I'm unable to access the "redis-replica" service as well as the "redis-master" service because Podman Desktop / Podman generates the following service:

```
apiVersion: v1
kind: Service
metadata:
  name: redis-replica-pod-35967
  namespace: default
spec:
  ports:
    - name: redis-replica-pod-35967
      protocol: TCP
      port: 35967
      targetPort: 6379
  selector:
    app: redis-replica-pod
```

When it should be creating this (example from a kompose conversion which works):

```
apiVersion: v1
kind: Service
metadata:
  name: redis-replica
spec:
  ports:
    - name: "6379"
      port: 6379
      targetPort: 6379
```

TLDR: When working with multiple containers with compose, it does not work well at all when converting to Kubernetes. 1 container is okay though.

_Originally posted by @cdrage in https://github.com/containers/podman-desktop/issues/3345#issuecomment-1803022622_
            

### Describe the solution you'd like

Improvements is:
- Port should be both (not just TCP)
- Port should be the same as target port so that it is accessible by the kubernetes pod. For example, within the compose yaml the frontend service wants to access redis-replica:6379. If deploying via podman containers it will be accessible. It should also still be able to access redis-replica:6379 when  it's in a pod. To do so a service needs to be made with the same name as well as port number. Right now, it is not.
- Name is random? It should be `redis-replica` not `redis-replica-pod-35967`

Take inspiration from kompose with regards to conversion practices for services.

**Comment by cdrage** [comment]
Still valid
